src/set_theory/game/nim.lean

```diff
@@ -0,0 +1,283 @@
+/-
+Copyright (c) 2020 Fox Thomson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fox Thomson
+-/
+import set_theory.game.impartial
+import set_theory.ordinal
+import data.set
+import logic.basic
+
+universes u v
+
+/-!
+# Nim and the Sprague-Grundy theorem
+
+This file contains the definition for nim for any ordinal `O`. In the game of `nim O₁` both players
+may move to `nim O₂` for any `O₂ < O₁`.
+We also define a Grundy value for an impartial game `G` and prove the Sprague-Grundy theorem, that
+`G` is equivalent to `nim (Grundy_value G)`.
+-/
+
+open pgame
+
+local infix ` ≈ ` := equiv
+
+/-- The definition of single-heap nim, which can be viewed as a pile of stones where each player can
+ take a positive number of stones from it on their turn. -/
+noncomputable def nim : ordinal → pgame
```

Comment:
I get a warning `definition 'nim' was incorrectly marked as noncomputable` here. Do you?

Reply:
However removing it gives the error `rec_fn_macro only allowed in meta definitions`... Sounds like it may be a Lean bug?

Reply:
I get this too, it should be noncomputable due to the `quotient.out`, right? Where would be best to flag this?

Reply:
Can you try to minimise this as an example? Ideally we could even get a MWE that doesn't even import mathlib, but it would suffice to get a MWE that just depends on the current mathlib. Can you try just copy and pasting this definition into a file that just imports `pgame`, `nth_rewrite` and `ordinal`? Check also if the `from` blocks can by replaced by sorry without making the problem go away.

After that, create a new issue here on github, and make a post on zulip to bring it to people's attention.

Reply:
(Unfortunately I think we better get to the bottom of this before merging, as we don't want warning in mathlib.)

Reply:
https://github.com/leanprover-community/lean/issues/451

Reply:
Mario has provided a workaround at https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/neither.20computable.20nor.20noncomputable.